golem-worker-service-base/src/gateway_execution/gateway_http_input_executor.rs

```diff
@@ -647,36 +601,178 @@ impl<Namespace: Send + Sync + Clone + 'static> GatewayHttpInputExecutor
 }
 
 async fn resolve_rib_input(
-    input: &serde_json::Map<String, Value>,
+    rich_request: &mut RichRequest,
```

Comment:
One thing I don't really like here is that we mutate the RichRequest (due to potentially moving the body).
Would it be possible to refactor this code so we move the request here? (So that it can not be used twice by accident)

Reply:
It's just like any server backend I guess - that once we consumes body. At the same time we don't need to access body if its not being used too. One thing that we could do is to populate body within RichRequest, once we access the body.